src/core/Value.cc

```diff
@@ -635,6 +635,61 @@ void VectorType::VectorObjectDeleter::operator()(VectorObject *v)
   delete orig;
 }
 
+std::shared_ptr<Values>
+VectorType::embeddedValues() const
+{
+  class VectorValueIterator: public VirtualValueIterator  {
+  private:
+    VectorType::iterator it;
+  public:
+    VectorValueIterator(VectorType::iterator i) : it(i) {};
+
+    void inc() { ++it; };
+    const Value& operator*() const { return *it; }
```

Comment:
In general when you override something, you want to mark it with `override`.

Reply:
I'm not used to that from my experience in Java and JavaScript, but it makes sense.